pkg/network/event_common.go

```diff
@@ -414,22 +414,25 @@ func printAddress(address util.Address, names []dns.Hostname) string {
 // This function generates all relevant combinations of connection keys: [(source, dest), (dest, source), (NAT'd source, NAT'd dest), (NAT'd dest, NAT'd source)].
 // This is necessary to handle all possible scenarios for connections originating from the USM module (i.e., whether they are NAT'd or not, and whether they use TLS).
 func ConnectionKeysFromConnectionStats(connectionStats ConnectionStats) []types.ConnectionKey {
-
+	hasTranslation := connectionStats.IPTranslation != nil
+	factor := 1
+	if hasTranslation {
+		factor = 2
+	}
+	connectionKeys := make([]types.ConnectionKey, 2*factor)
 	// USM data is always indexed as (client, server), but we don't know which is the remote
 	// and which is the local address. To account for this, we'll construct 2 possible
 	// connection keys and check for both of them in the aggregations map.
-	connectionKeys := []types.ConnectionKey{
-		types.NewConnectionKey(connectionStats.Source, connectionStats.Dest, connectionStats.SPort, connectionStats.DPort),
-		types.NewConnectionKey(connectionStats.Dest, connectionStats.Source, connectionStats.DPort, connectionStats.SPort),
-	}
+	connectionKeys[0] = types.NewConnectionKey(connectionStats.Source, connectionStats.Dest, connectionStats.SPort, connectionStats.DPort)
+	connectionKeys[1] = types.NewConnectionKey(connectionStats.Dest, connectionStats.Source, connectionStats.DPort, connectionStats.SPort)
 
 	// if IPTranslation is not nil, at least one of the sides has a translation, thus we need to add translated addresses.
 	if connectionStats.IPTranslation != nil {
 		localAddress, localPort := GetNATLocalAddress(connectionStats)
 		remoteAddress, remotePort := GetNATRemoteAddress(connectionStats)
-		connectionKeys = append(connectionKeys,
-			types.NewConnectionKey(localAddress, remoteAddress, localPort, remotePort),
-			types.NewConnectionKey(remoteAddress, localAddress, remotePort, localPort))
+		connectionKeys[2] = types.NewConnectionKey(localAddress, remoteAddress, localPort, remotePort)
```

Comment:
can you please add documentation about this part, it is pretty confusing to understand

Reply:
there is the original comment in line 429
what is missing?

Reply:
you are now conceding the index at connectionKeys while in the past it was only appends, so maybe add a short explanation about the ability to treat each index and its meaning, if you think it is redundant feel free to resolve this comment :)

Reply:
I'm allocating the full buffer before, and set the len to be 2 (or 4) instead of 0
so I already have entries
I can restore the logic to allocate with len == 0, and then have appends
but that's the same